src/openrct2/world/footpath.c

```diff
@@ -270,7 +270,8 @@ static money32 footpath_element_insert(sint32 type, sint32 x, sint32 y, sint32 z
 
 static money32 footpath_element_update(sint32 x, sint32 y, rct_tile_element *tileElement, sint32 type, sint32 flags, uint8 pathItemType)
 {
-    if (footpath_element_get_type(tileElement) != (type & 0x0F) || (tileElement->type & FOOTPATH_ELEMENT_TYPE_FLAG_IS_QUEUE) != (type >> 7)) {
+    if (footpath_element_get_type(tileElement) != (type & (FOOTPATH_PROPERTIES_TYPE_MASK >> 4)) ||
```

Comment:
This looks a bit dirty to me. If several values are put together in `type`, perhaps we should split those?